src/ray/raylet/node_manager.cc

```diff
@@ -475,95 +487,73 @@ void NodeManager::ProcessClientMessage(
     SubmitTask(task, Lineage());
   } break;
   case protocol::MessageType::ReconstructObjects: {
-    // TODO(hme): handle multiple object ids.
     auto message = flatbuffers::GetRoot<protocol::ReconstructObjects>(message_data);
+    std::vector<ObjectID> required_object_ids;
     for (size_t i = 0; i < message->object_ids()->size(); ++i) {
       ObjectID object_id = from_flatbuf(*message->object_ids()->Get(i));
-      RAY_LOG(DEBUG) << "reconstructing object " << object_id;
       if (!task_dependency_manager_.CheckObjectLocal(object_id)) {
-        // TODO(swang): Instead of calling Pull on the object directly, record the
-        // fact that the blocked task is dependent on this object_id in the task
-        // dependency manager.
-        RAY_CHECK_OK(object_manager_.Pull(object_id));
+        if (message->fetch_only()) {
+          RAY_CHECK_OK(object_manager_.Pull(object_id));
```

Comment:
Don't we want to `Pull` in the non-fetch-only case as well?

I interpreted the `fetch_only` flag as "if true, then just fetch, if false, then fetch + reconstruct".

Reply:
Oops, thanks!

Reply:
Ah actually, the reason I did that was because the task dependency manager calls `Pull` internally once it determines that a dependency is missing. A little unfortunate, but I guess this makes sense for now while we have the `fetch_only` flag? I'll leave a comment about that.